Update RFC README.md (#16944)
microsoft/fluentui: rfcs/README.md

We're trying a new approach of checking in Fluent UI RFCs to the repo so that they're more persistent, easier to find, and we can use PR comments to discuss the proposals.

## Why RFCs?

The RFC process is intended to bring a few benefits:

- More efficent and durable decision making
  - **Efficent** as we can all engage and review on our own time, rather than locking X engineers in a room for hours
  - **Durable** as we give more time for decisions to be reviewed, commented, and considered. Rather than making a decision

>     after being locked in a room for an hour with X other engineers.

Reviewer: nit: These two should go in the same line.

Author: fixed

- A more inclusive review proces
  - Supporting both quick and deep thinkers
  - Allowing proposals and comments from contributors
- Capturing decisions, allowing:
  - Us to review recent decisions and build ontop of them
  - Others to join the project more easily

The React RFC Process sums it up nicely[1]

> The RFC process is a great opportunity to get more eyeballs on your proposal before it becomes a part of a released version. Quite often, even proposals that seem "obvious" can be significantly improved once a wider group of interested people have a chance to weigh in.
>
> The RFC process can also be helpful to encourage discussions about a proposed feature as it is being designed, and incorporate important constraints into the design while it's easier to change, before the design has been fully implemented.

## RFC process

### 1. Creation

- Start by making a copy of [TEMPLATE.md](./TEMPLATE.md) under an appropriate area sub-folder and name. (While we're still experimenting with the RFC process, feel free to create a new area sub-folder if you don't think an appropriate one exists yet.)
- Write the RFC
- Submit a PR and label it with **Type: RFC**.

### 2. Review

> A key purpose of the review phase is to seek feedback to help improve your idea. You the author are ultimately accountable for the proposal and it's impact. Reviewers can, through comments, help you make the best decision you can. Throughout this review and process it is important that we strive to avoid both [Design by Committee](https://en.wikipedia.org/wiki/Design_by_committee) and [Diffusion of Responsibility](https://en.wikipedia.org/wiki/Diffusion_of_responsibility).

Reviewer 2: ```suggestion
A key purpose of the review phase is to seek feedback to help improve your idea. You the author are ultimately accountable for the proposal and its impact. Reviewers can, through comments, help you make the best decision you can. Throughout this review and process it is important that we strive to avoid both [Design by Committee](https://en.wikipedia.org/wiki/Design_by_committee) and [Diffusion of Responsibility](https://en.wikipedia.org/wiki/Diffusion_of_responsibility).
```

> - Seek support and integrate feedback.

Reviewer 2: One thing I wonder if we should recommend specifically/only for RFCs is to use single comments rather than reviews with multiple comments--even though that's opposite of what I'd normally recommend/what I often do. The reason is I've noticed with the long, complex comment threads that can arise on RFCs, when some of the comments came as part of a review, the threads become extremely hard to follow (thanks to github's UI being terrible). In some cases you can reasonably find/read all the threads in the files view, but that doesn't work if a thread has disappeared from files view due to new changes touching that line.

Author: I'm not sure I understand the issue fully. 

Is this saying that when people use the "Start a Review" feature then sometimes the comment threads get broken? I can't say I've seen this happen so far. 

I'm nervous to start baking in guidance here that is specifically working around a particular behavior of the tools. Though again i'm not sure the impact of this particular issue.

Reviewer 2: Yes that's what I meant. It's more as a suggestion than required guidance, but it's based on my experience with comment threads on my RFCs that became really hard to follow when some or all of the comments came in as part of larger reviews.

Reviewer 2: Also since github PR reviews are the tool we're planning to use for RFCs, and the review interface is known to be far less than ideal in some ways, I think it's worth mentioning suggested mitigations in official guidance. (though in general I agree about avoiding guidance specific to tools that aren't an inherently required part of a process)

Author: Got it. Let's hold off for now. If it becomes a significant issue or if we have the need for other tips we can add it in.

  - It is the responsibility of the author(s) to read, consider, and respond to all feedback in a timely manner. This does not mean all feedback must be integrated into the proposal, but it must be considered.

>   - If you want to good engagement on your RFC, reserve time to respond to comments quickly.

Reviewer: ```suggestion
  - If you want to have good engagement on your RFC, reserve time to respond to comments quickly.
```

> Once the RFC has been open for **at least 2 days**, and recieved sufficent comments, the author can begin seeking approval.

Reviewer 3: **suggestion:**
- Create Draft PR
- gather initial comments and incorporate suggestions if applicable
- promote  PR to "Ready for review" aka ready for seeking approval

Author: Good suggestion. Maybe a label "Ready for final review". 

I was leaving it a little open and just saying "seeking approval" as there likely will be an aspect of actively going out and asking for approval. Tho having a label to mark it ready wouldn't change this and would only help.

Reviewer 3: yeah, I mean label === PR Draft , no need for yet another label ? :D

Author: Ohh, ok. Yeah spinning up an RFC as a Draft would be fine and a good way to get early feedback.

I don't think we should require all RFCs to start as Draft. Just like a normal PR folks should use the Draft state to communicate that this is not ready for broad feedback. Also, just like a normal PR it's totally fine to reach out to specific people and ask them to weigh in on the PR early. 

I'll add a line talking about this.

I do think we should just assume that a non-draft PR is ready for broad review.

> Once the RFC has been open for **at least 2 days**, and recieved sufficent comments, the author can begin seeking approval.

Reviewer: ```suggestion
Once the RFC has been open for **at least 2 days**, and received sufficient comments, the author can begin seeking approval.
```

### 3. Approval

We're still working through details for approval, below are the current guidelines.

Following the theme of author accountability, in general it's up to the author to decide when they feel they have recieved enough feedback, and enough approvals to proceed.

- When in doubt, seek **2 approvals** before merging the RFC.

> ### 4. Implementation

Reviewer 2: Might be worth mentioning that (while this practice isn't recommended) if implementation has already started on any parts of the RFC, that should be noted in the RFC text.

Author: Yep, it may, but let's keep an eye on whether this becomes a recurring issue. Same as what I said on the comment thread before, if this is a pattern we can update guidance, but let's not overreact and add on to the guidance based on just one instance (also adding guidance here may end up inviting implementation to start before the RFC is complete which is kind of a slippery slope)

> Following the theme of author accountability, in general it's up to the author to decide when they feel they have recieved enough feedback, and enough approvals to proceed.

Reviewer: ```suggestion
Following the theme of author accountability, in general it's up to the author to decide when they feel they have received enough feedback, and enough approvals to proceed.
```

> - When in doubt, seek **2 approvals** before merging the RFC.

Reviewer: ```suggestion
- When in doubt, seek **at least 2 approvals** before merging the RFC.
```

We're trying a new approach of checking in Fluent UI RFCs to the repo so that they're more persistent, easier to find, and we can use PR comments to discuss the proposals.

## Why RFCs?

The RFC process is intended to bring a few benefits:

- More efficent and durable decision making
  - **Efficent** as we can all engage and review on our own time, rather than locking X engineers in a room for hours
  - **Durable** as we give more time for decisions to be reviewed, commented, and considered. Rather than making a decision
    after being locked in a room for an hour with X other engineers.
- A more inclusive review proces
  - Supporting both quick and deep thinkers
  - Allowing proposals and comments from contributors
- Capturing decisions, allowing:
  - Us to review recent decisions and build ontop of them
  - Others to join the project more easily

> The React RFC Process sums it up nicely[1]

Reviewer 2: I think it needs to be like this for the link to work
```suggestion
The [React RFC Process][1] sums it up nicely
```

Author: Actually looks like it needs to be:
```
The React RFC Process<sup>[1](#f1)</sup> sums it up nicely
...
- <b id="f1">1</b> - https://github.com/reactjs/rfcs/blob/master/README.md
```

Reviewer 2: Do we really need to render it as a footnote though? Seems like just a normal link would be enough.

Author: You're right a link should be preferred, though imo a footnote would be OK as well. I did update it to a link though.
